string key
float32 values
